### PR TITLE
feat: update sdk with recent changes\n\nBREAKING CHANGE: \n\nDescription of the breaking change

### PR DESCRIPTION
**Pull request by itsdebs**
## Description of the change

> Updated fork with changes from base

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by jdreamerz — COMMENTED** [review]
I don't have a lot of context here, but some comments on minor things I did have questions about.

File: .buildscripts/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# #!/bin/sh
```
Comment (bardisg) [inline .buildscripts/e2e.sh:1]: delete this fle, we do not use it

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version: 2
```
Comment (bardisg) [inline .github/dependabot.yml:1]: also add thecheck for github actons

**Comment by bardisg** [comment]
@debanjan97 please also add the changelog for this PR in History.md or delete it and create a CHANGELOG.md file with it

File: .gitignore [hunk 3]
```diff
@@ -6,6 +6,9 @@ dist
 MANIFEST
 build
 .eggs
-[metadata]
+*.bat
+.vscode/
+.idea/
+.python-version
 **/.DS_Store
-.vscode/launch.json
```
[the remark this answers is not in the document]
Comment (itsdebs) [reply .gitignore:11]: This is a change from the forked repo, we have .`vscode/` in .gitignore, I think will cover this.

File: .gitignore [hunk 4]
```diff
@@ -6,6 +6,9 @@ dist
 MANIFEST
 build
 .eggs
-[metadata]
```
[the remark this answers is not in the document]
Comment (itsdebs) [reply .gitignore:9]: This is a change from the forked repo, Please let me know if this needs to change.

File: rudderstack/analytics/client.py [hunk 5]
```diff
@@ -225,37 +249,38 @@ def _enqueue(self, msg):
...
         msg['anonymousId'] = stringify_id(msg.get('anonymousId', None))
-
+        msg['channel'] = CHANNEL
         msg = clean(msg)
         self.log.debug('queueing: %s', msg)
 
+        # Check message size.
+        msg_size = len(json.dumps(msg, cls=DatetimeSerializer).encode())
+        if msg_size > MAX_MSG_SIZE:
+            raise RuntimeError('Message exceeds %skb limit. (%s)', str(int(MAX_MSG_SIZE / 1024)), str(msg))
```
[the remark this answers is not in the document]
Comment (itsdebs) [reply rudderstack/analytics/client.py:274]: `f'Message exceeds {(MAX_MSG_SIZE / 1024):.0f}kb limit. {msg}'`
Is this the correct way?

File: rudderstack/analytics/test/test_client.py [hunk 6]
```diff
@@ -249,8 +248,8 @@ def test_flush(self):
     def test_shutdown(self):
         client = self.client
         # set up the consumer with more requests than a single batch will allow
-        for i in range(1000):
-            success, msg = client.identify('userId', {'trait': 'value'})
+        for _ in range(1000):
```
[the remark this answers is not in the document]
Comment (itsdebs) [reply rudderstack/analytics/test/test_client.py:251]: This is from forked repo and for test purpose, I have ignored this to reduce conflicts on next merge.

File: setup.py [hunk 7]
```diff
@@ -56,16 +57,9 @@
...
         "Programming Language :: Python",
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.6",
-        "Programming Language :: Python :: 2.7",
-        "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.2",
-        "Programming Language :: Python :: 3.3",
-        "Programming Language :: Python :: 3.4",
-        "Programming Language :: Python :: 3.5",
         "Programming Language :: Python :: 3.6",
```
[the remark this answers is not in the document]
Comment (itsdebs) [reply setup.py:60]: any version < 3.6

**Comment by bardisg** [comment]
can we fix the 2 bugs reported in sonar https://sonarcloud.io/project/issues?resolved=false&types=BUG&sinceLeakPeriod=true&pullRequest=15&id=rudderlabs_rudder-sdk-python please

File: simulator.py [hunk 8]
```diff
@@ -81,6 +94,10 @@ def unknown():
 analytics.debug = True
 analytics.dataPlaneUrl = options.dataPlaneUrl
 
+analytics.gzip = options.gzip    
+
+print(analytics.gzip)
```
Comment (bardisg) [inline simulator.py:99]: is this print valid here?

File: rudderstack/analytics/client.py [hunk 9]
```diff
@@ -248,6 +250,10 @@ def _enqueue(self, msg):
         if message_id is None:
             message_id = uuid4()
 
+        # default integrations should be "All": True
+        if msg['integrations'] == {} :
```
Comment (bardisg) [inline rudderstack/analytics/client.py:254]: Should this apply for all server side sdks?

**Review by bardisg — APPROVED** [review]

**Review by bardisg — CHANGES_REQUESTED** [review]
Tests are failing seems to be an issue with a key name traits vs trait

File: rudderstack/analytics/client.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jdreamerz) [reply rudderstack/analytics/client.py:279]: Looks reasonable.

File: rudderstack/analytics/client.py (hunk 9 above)
Comment (bardisg) [reply rudderstack/analytics/client.py:254]: verified that thsi is not required

**Comment by bardisg** [comment]
all my comments are addressed. unable to resolve the discussion unsure of the reason

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
```
Comment (bardisg) [inline CHANGELOG.md:3]: changelog and histroy md should match and also mention as breaking change that support for 3.6 <  is dropped

File: rudderstack/analytics/request.py [hunk 11]
```diff
@@ -1,29 +1,44 @@
...
 
-def post(write_key, host=None, timeout=15, **kwargs):
+def post(write_key, host=None, gzip=True, timeout=15, proxies=None, **kwargs):
     """Post the `kwargs` to the API"""
-    log = logging.getLogger('rudder')
+    log = logging.getLogger('rudderstack')
     body = kwargs
     body["sentAt"] = datetime.utcnow().replace(tzinfo=tzutc()).isoformat()
-    url = remove_trailing_slash(host or 'https://hosted.rudderlabs.com') + '/v1/batch'
+    url = remove_trailing_slash(host or 'https://api.rudderstack.io') + '/v1/batch'
```
Comment (bardisg) [inline rudderstack/analytics/request.py:21]: this url is wrong rudderstack.io not existing

**Review by bardisg — APPROVED** [review]

File: rudderstack/analytics/request.py (hunk 11 above)
Comment (itsdebs) [reply rudderstack/analytics/request.py]: changed to api.rudderstack.com

**Pull request merged**